controller/src/main/java/io/pravega/controller/store/stream/records/ReaderGroupConfigRecord.java

```diff
@@ -52,10 +53,10 @@
     public static ReaderGroupConfigRecord update(ReaderGroupConfig rgConfig, long generation, boolean isUpdating) {
         Map<String, RGStreamCutRecord> startStreamCuts = rgConfig.getStartingStreamCuts().entrySet().stream()
                 .collect(Collectors.toMap(e -> e.getKey().getScopedName(),
-                        e -> new RGStreamCutRecord(getStreamCutMap(e.getValue()))));
+                        e -> new RGStreamCutRecord(ImmutableMap.copyOf(ModelHelper.getStreamCutMap(e.getValue())))));
```

Comment:
can we make modelhelper return an imutable map itself. rather than create another copy here.

Reply:
Another method that uses getStreamCutMap requires a Map and not an Immutable map.
Given that size of this map should be pretty small typically < 10 it should be fine to do the conversion.

Reply:
ImmutableMap extends map. so in model helper you can construct an immutable map itself.

Reply:
done in PR5466